GeneratorInterface/Pythia8Interface/plugins/JetMatchingEWKFxFx.cc

```diff
@@ -0,0 +1,998 @@
+      // Append unmatched hadronic jets
+      tempEventJet.append(ID_GLUON, 98, 0, 0, 0, 0, 0, 0, pIn.px(), pIn.py(), pIn.pz(), pIn.e());
+    }
+
+    Vec4 pIn = eventProcess[typeIdx[0][iNow]].p();
+    // Append the current parton
+    tempEventJet.append(ID_GLUON, 99, 0, 0, 0, 0, 0, 0, pIn.px(), pIn.py(), pIn.pz(), pIn.e());
+    if (!slowJet->setup(tempEventJet)) {
+      errorMsg(
+          "Warning in JetMatchingMadgraph:matchPartonsToJets"
+          "Light: the SlowJet algorithm failed on setup");
+      return NONE;
+    }
+    // These are the conditions for the hadronic jet to match the parton
+    //  at the local qCut scale
+    if (slowJet->iNext() == tempEventJet.size() - 1 && slowJet->jNext() > -1 && slowJet->dNext() < localQcutSq) {
+      int iKnt = -1;
+      for (int i = 0; i != tempSize; ++i) {
+        if (jetAssigned[i])
+          continue;
+        ++iKnt;
+        // Identify the hadronic jet that matches the parton
+        if (iKnt == slowJet->jNext())
+          jetAssigned[i] = true;
+      }
+    } else {
+      return UNMATCHED_PARTON;
+    }
+    ++iNow;
+  }
+  // Minimal eT/pT (CellJet/SlowJet) of matched light jets.
+  // Needed later for heavy jet vetos in inclusive mode.
+  // This information is not used currently.
+  if (nParton > 0 && pTminEstimate > 0)
+    eTpTlightMin = pTminEstimate;
+  else
+    eTpTlightMin = -1.;
+
+  // Record the jet separations.
+  setDJR(workEventJet);
+
+  // No veto
+  return NONE;
+}
+
+int JetMatchingEWKFxFx::matchPartonsToJetsHeavy() {
+  // Currently, heavy jets are unmatched
+  // If there are no extra jets, then accept
+  // jetMomenta is NEVER used by MadGraph and is always empty.
+  //  This check does nothing.
+  //  Rather, if there is any heavy flavor that is harder than
+  //  what is present at the LHE level, then the event should
+  //  be vetoed.
+
+  // if (jetMomenta.empty()) return NONE;
+  // Count the number of hard partons
+  int nParton = typeIdx[1].size();
+
+  Event tempEventJet(workEventJet);
+
+  double scaleF(1.0);
+  // Rescale the heavy partons that are from the hard process to
+  //  have pT=collider energy.   Soft/collinear gluons will cluster
+  //  onto them, leaving a remnant of hard emissions.
+  for (int i = 0; i < nParton; ++i) {
+    scaleF = eventProcessOrig[0].e() / workEventJet[typeIdx[1][i]].pT();
+    tempEventJet[typeIdx[1][i]].rescale5(scaleF);
+  }
+
+  if (!hjSlowJet->setup(tempEventJet)) {
+    errorMsg(
+        "Warning in JetMatchingMadgraph:matchPartonsToJets"
+        "Heavy: the SlowJet algorithm failed on setup");
+    return NONE;
+  }
+
+  while (hjSlowJet->sizeAll() - hjSlowJet->sizeJet() > 0) {
+    if (hjSlowJet->dNext() > qCutSq)
+      break;
+    hjSlowJet->doStep();
+  }
+
+  int nCLjets(0);
+  // Count the number of clusters with pT>qCut.  This includes the
+  //  original hard partons plus any hard emissions.
+  for (int idx = 0; idx < hjSlowJet->sizeAll(); ++idx) {
+    if (hjSlowJet->pT(idx) > sqrt(qCutSq))
```

Comment:
```suggestion
    if (hjSlowJet->pT(idx) > qCut)
```
(or is it possible that `qCut` assumes negative values?)

Reply:
In principle qCut should not take negative values so indeed this does can be done.

Reply:
@Cvico this was not updated with the latest commit...

Reply:
please commit now